src/createSchemaForm.tsx

```diff
@@ -563,21 +563,39 @@ export function createTsForm<
     }
 
     const renderedFields = renderFields(schema, props);
-    const renderedFieldNodes = flattenRenderedElements(renderedFields);
     return (
       <FormProvider {..._form}>
         <ActualFormComponent {...formProps} onSubmit={submitFn}>
           {renderBefore && renderBefore({ submit: submitFn })}
-          {CustomChildrenComponent ? (
-            <CustomChildrenComponent {...renderedFields} />
-          ) : (
-            renderedFieldNodes
-          )}
+          <Children
+            renderedFields={renderedFields}
+            CustomChildrenComponent={CustomChildrenComponent}
+          />
+
           {renderAfter && renderAfter({ submit: submitFn })}
         </ActualFormComponent>
       </FormProvider>
     );
   };
+
+  // these needs to at least have one component wrapping it or the context won't propogate
+  // i believe that means any hooks used in the CustomChildrenComponent are really tied to the lifecycle of this Children component... 😬
+  // i ~think~ that's ok
+  function Children<SchemaType extends RTFFormSchemaType>({
+    CustomChildrenComponent,
+    renderedFields,
+  }: {
+    renderedFields: RenderedFieldMap<SchemaType>;
+    CustomChildrenComponent?: FunctionComponent<RenderedFieldMap<SchemaType>>;
```

Comment:
is this really a `FunctionComponent`? isn't it just a function? it's a render prop not a component yeah?

Reply:
Oh good point! That's the whole reason for this change lol :)